checker.cpp

```diff
@@ -3,44 +3,55 @@
 #include <unistd.h>
 using namespace std;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if(temperature > 102 || temperature < 95) {
-    cout << "Temperature critical!\n";
+void printPattern() {
     for (int i = 0; i < 6; i++)
     {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+        cout << "\r* " << flush;
+         sleep(1);
+         cout << "\r *" << flush;
+         sleep(1);
     }
+}
 
-    return 0;
-  } else if(pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+bool isTemperatureCritical(float temperature) {
+    bool result = false;
+    result = temperature > 102 || temperature < 95;
+    if (result == true) {
+        cout << "Temperature critical!\n";
+        printPattern();
     }
-    return 0;
-  } else if(spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+
+    return result;
+}
+
+bool isPulseRateWithinRange(float pulseRate) {
+    bool result = false;
+    result = pulseRate < 60 || pulseRate > 100;
```

Comment:
consider isolating range comparison in a separate function to make it easier to implement warnings